### PR TITLE
Flaky test 2 14 x

### PR DESCRIPTION
**Pull request by EugenioCollado**
## Description

Since update in CI to new Windows version, the test `BlackboxTests_DDS_PIM.DDSStatus.sample_lost_be_dw_be_dr_fragments` is failing consistently (it was flaky before the update). This PR tries to fix this by increasing the time btw consecutive writes.

## Contributor Checklist

- [ ] Commit messages follow the project guidelines. 
- [ ] The code follows the style guidelines of this project. 
- _N/A_ Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- _N/A_ Any new/modified methods have been properly documented using Doxygen. 
- _N/A_ Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- [ ] Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- [ ] Changes are API compatible. 
- _N/A_ New feature has been added to the `versions.md` file (if applicable).
- _N/A_ New feature has been documented/Current behavior is correctly described in the documentation. 
    
- _N/A_ Applicable backports have been included in the description.

## Reviewer Checklist

- [ ] The PR has a milestone assigned.
- [ ] The title and description correctly express the PR's purpose.
- [ ] Check contributor checklist is correct.
- [ ] If this is a critical bug fix, backports to the critical-only supported branches have been requested.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/rtps/transport/asio_helpers.hpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ struct asio_helpers
         assert(initial_buffer_value >= minimum_buffer_value);
 
         final_buffer_value = initial_buffer_value;
-        while (final_buffer_value > minimum_buffer_value)
+        while (final_buffer_value >= minimum_buffer_value)
```
Comment (MiguelCompany) [inline src/cpp/rtps/transport/asio_helpers.hpp]: No need for this, since a final attempt with `minimum_buffer_value` is performed after the while loop is finished.

File: test/blackbox/api/fastrtps_deprecated/PubSubReader.hpp [hunk 2]
```diff
@@ -1102,10 +1102,10 @@ class PubSubReader
     }
 
     PubSubReader& socket_buffer_size(
-            uint32_t sockerBufferSize)
+            uint32_t socketBufferSize)
```
Comment (MiguelCompany) [inline test/blackbox/api/fastrtps_deprecated/PubSubReader.hpp:1105]: Same typo exists in dds-pim

File: test/blackbox/common/DDSBlackboxTestsListeners.cpp [hunk 3]
```diff
@@ -776,7 +776,13 @@ void sample_lost_test_dr_init(
         PubSubReader<T>& reader,
         std::function<void(const eprosima::fastdds::dds::SampleLostStatus& status)> functor)
 {
-    reader.sample_lost_status_functor(functor)
+    auto testTransport = std::make_shared<test_UDPv4TransportDescriptor>();
+    testTransport->sendBufferSize = SAMPLE_LOST_TEST_BUFFER_SIZE;
+    testTransport->receiveBufferSize = SAMPLE_LOST_TEST_BUFFER_SIZE;
+
+    reader.disable_builtin_transport()
+            .add_user_transport_to_pparams(testTransport)
```
Comment (MiguelCompany) [inline test/blackbox/common/DDSBlackboxTestsListeners.cpp]: Is this really necessary?
I think `reader.socket_buffer_size(SAMPLE_LOST_TEST_BUFFER_SIZE)` would be enough.

File: test/blackbox/common/DDSBlackboxTestsListeners.cpp [hunk 4]
```diff
@@ -788,9 +794,6 @@ void sample_lost_test_init(
         PubSubWriter<T>& writer,
         std::function<void(const eprosima::fastdds::dds::SampleLostStatus& status)> functor)
 {
-    reader.socket_buffer_size(SAMPLE_LOST_TEST_BUFFER_SIZE);
```
Comment (MiguelCompany) [inline test/blackbox/common/DDSBlackboxTestsListeners.cpp:791]: ... so I think we can keep the line for the reader.

The line for the writer does not harm, but I am ok with removing it.

File: src/cpp/rtps/transport/asio_helpers.hpp (hunk 1 above)
Comment (EugenioCollado) [reply src/cpp/rtps/transport/asio_helpers.hpp]: But in that final attempt, the condition `(option.value() >= value_to_set) && (option.value() <= max_value)` is checked, which can fail if the socket buffer size actually set is higher than the requested value. For example, in many tests the buffer size is requested as 1024 and the max message size is also 1024. This corner case causes the code to jump directly to that final check, and since `socket.set_option` often sets a value around 2K, the method fails because that value exceeds ´max_value´ (´max_value = static_cast<int32_t>(initial_buffer_value)´).

A better solution might be to remove the upper-limit check against `max_value`, since that logic isn’t applied in the main while loop.

File: test/blackbox/common/DDSBlackboxTestsListeners.cpp (hunk 3 above)
Comment (EugenioCollado) [reply test/blackbox/common/DDSBlackboxTestsListeners.cpp]: You are right, this is a hot fix done when the builtin transports buffer size was not being correctly set.

File: test/blackbox/common/DDSBlackboxTestsListeners.cpp (hunk 4 above)
Comment (EugenioCollado) [reply test/blackbox/common/DDSBlackboxTestsListeners.cpp:791]: I think it’s best to remove that line, as it could lead future readers to believe the entity is using built-in transports.

File: src/cpp/rtps/transport/asio_helpers.hpp (hunk 1 above)
Comment (MiguelCompany) [reply src/cpp/rtps/transport/asio_helpers.hpp]: > A better solution might be to remove the upper-limit check against `max_value`, since that logic isn’t applied in the main while loop.

This is better IMO.

File: test/blackbox/common/DDSBlackboxTestsListeners.cpp (hunk 4 above)
Comment (MiguelCompany) [reply test/blackbox/common/DDSBlackboxTestsListeners.cpp:791]: Might be also worth moving the line for the reader inside `sample_lost_test_dr_init`

**Comment by EugenioCollado** [comment]
@MiguelCompany all suggestions applied. The CI fail is unrelated as it is about the `test/blackbox/types/Data100kb.cxx` which is not even modified in this PR

File: src/cpp/rtps/transport/asio_helpers.hpp [hunk 5]
```diff
@@ -86,11 +86,9 @@ struct asio_helpers
         socket.set_option(BufferOptionType(value_to_set), ec);
         if (!ec)
         {
-            // Last attempt was successful. Get the actual value set.
```
Comment (MiguelCompany) [inline src/cpp/rtps/transport/asio_helpers.hpp:89]: Please leave the comment

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
Just one NIT. Does not need to run CI again

File: src/cpp/rtps/transport/asio_helpers.hpp (hunk 5 above)
Comment (EugenioCollado) [reply src/cpp/rtps/transport/asio_helpers.hpp:89]: Done in [97a6472](https://github.com/eProsima/Fast-DDS/pull/5921/commits/97a6472bb56ff9eb8895b4a5541c0bb605bd12a2)

**Review by MiguelCompany — APPROVED** [review]

**Comment by MiguelCompany** [comment]
CI passed for 759e5f96d1637962069ec4402594dc10ab271684
